mico-core/src/main/java/io/github/ust/mico/core/service/GitHubCrawler.java

```diff
@@ -53,48 +57,96 @@ public GitHubCrawler(RestTemplate restTemplate, KubernetesNameNormalizer kuberne
         this.kubernetesNameNormalizer = kubernetesNameNormalizer;
     }
 
-    private MicoService crawlGitHubRepo(String uriBasicInfo, String uriReleaseInfo) throws IOException {
+    private MicoService crawlGitHubRepo(String uriBasicInfo, String uriReleaseInfo, String dockerfilePath) throws IOException {
         log.debug("Crawl GitHub basic information from '{}' and release information from '{}'", uriBasicInfo, uriReleaseInfo);
 
-        ResponseEntity<String> responseBasicInfo = restTemplate.getForEntity(uriBasicInfo, String.class);
-        ResponseEntity<String> responseReleaseInfo = restTemplate.getForEntity(uriReleaseInfo, String.class);
-        ObjectMapper mapper = new ObjectMapper().configure(DeserializationFeature.FAIL_ON_UNKNOWN_PROPERTIES, false);
+        ResponseEntity<String> responseBasicInfo;
+        ResponseEntity<String> responseReleaseInfo;
+
+        try {
+            responseBasicInfo = restTemplate.getForEntity(uriBasicInfo, String.class);
+        } catch (Exception e) {
```

Comment:
You should not catch a generic exception and throw a more specific one. For example, this catches an IOException if the network is down and it looks like that the repository does not exist.